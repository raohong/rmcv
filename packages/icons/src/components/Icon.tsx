import classnames from 'classnames';
import React from 'react';
import type { RootIconProps } from '../interface';

const Icon = (
  { component: Com, className, color, style, size, ...rest }: RootIconProps,
  ref: React.Ref<HTMLSpanElement>,
) => {
  return (
    <span
      className={classnames('rmc-vant-icon', className)}
      role="img"
      ref={ref}
      style={{
        color,
        fontSize: size,
        ...rest,
      }}
      {...rest}
    >
      <Com width="1em" height="1em" fill="currentColor" {...rest} />
    </span>
  );
};

export default React.forwardRef(Icon);
