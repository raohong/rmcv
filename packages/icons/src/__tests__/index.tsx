import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Icon, { Arrow } from '..';

const prefixCls = 'rmc-vant-icon';

const CustomSvgComponent: React.FC = (props) => (
  <svg
    viewBox="0 0 1024 1024"
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
    p-id="3356"
    data-testid="svg"
    {...props}
  >
    <path
      d="M682.666667 554.666667V213.333333H256v341.333334a85.333333 85.333333 0 0 0 85.333333 85.333333h256a85.333333 85.333333 0 0 0 85.333334-85.333333zM213.333333 128h640a85.333333 85.333333 0 0 1 85.333334 85.333333v128a85.333333 85.333333 0 0 1-85.333334 85.333334h-85.333333v128a170.666667 170.666667 0 0 1-170.666667 170.666666H341.333333a170.666667 170.666667 0 0 1-170.666666-170.666666V170.666667a42.666667 42.666667 0 0 1 42.666666-42.666667z m554.666667 85.333333v128h85.333333V213.333333h-85.333333zM128 810.666667h682.666667a42.666667 42.666667 0 0 1 0 85.333333H128a42.666667 42.666667 0 0 1 0-85.333333z"
      p-id="3357"
    />
  </svg>
);

test('render correctly', () => {
  const tree = render(<Arrow />);

  expect(tree.asFragment()).toMatchSnapshot();
});

test('render builtin icon', () => {
  render(<Arrow />);

  const com = screen.getByRole('img');

  expect(com.tagName).toBe('SPAN');
  expect(com.className).toBe(prefixCls);
});

test('render with className', () => {
  render(<Arrow className="test" />);

  expect(screen.getByRole('img')).toHaveClass('test');
});

test('render custom icon', () => {
  render(<Icon component={CustomSvgComponent} />);

  expect(screen.getByTestId('svg')).toHaveAttribute('width', '1em');
});
