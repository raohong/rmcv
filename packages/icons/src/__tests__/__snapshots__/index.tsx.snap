// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  <span
    aria-label="Arrow"
    class="rmc-vant-icon"
    role="img"
  >
    <svg
      fill="currentColor"
      height="1em"
      version="1.1"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M728.223744 520.22784a42.467328 42.467328 0 0 1-11.393024 20.503552L374.90688 882.65728c-16.662528 16.662528-43.677696 16.662528-60.340224 0s-16.662528-43.677696 0-60.340224L626.449408 510.43328 314.614784 198.598656c-16.662528-16.662528-16.662528-43.677696 0-60.340224 16.661504-16.662528 43.676672-16.662528 60.3392 0L716.879872 480.18432c10.860544 10.860544 14.642176 26.120192 11.343872 40.04352z"
      />
    </svg>
  </span>
</DocumentFragment>
`;
